js/private/js_binary.bzl

```diff
@@ -575,13 +575,10 @@ def _js_binary_impl(ctx):
     providers = []
 
     if ctx.attr.testonly and ctx.configuration.coverage_enabled:
-        # We have to instruct rule implementers to have this attribute present.
-        if not hasattr(ctx.attr, "_lcov_merger"):
-            fail("_lcov_merger attribute is missing and coverage was requested")
-
         # We have to propagate _lcov_merger runfiles since bazel does not treat _lcov_merger as a proper tool.
         # See: https://github.com/bazelbuild/bazel/issues/4033
-        runfiles = runfiles.merge(ctx.attr._lcov_merger[DefaultInfo].default_runfiles)
+        if hasattr(ctx.attr, "_lcov_merger"):
```

Comment:
Lets add a comment explaining why this is now optional, maybe with a link to whatever commit/issue explains it?

Will there be any reason for this in bazel8+? Should we add `# TODO: remove once bazel<8 support dropped`?

Reply:
Great 👍